lib/registry.js

```diff
@@ -108,6 +110,9 @@ function Registry (options) {
   if (HIVES.indexOf(_hive) == -1)
     throw new Error('illegal hive specified.');
 
+  if (_arch && _arch !== 64 && _arch !== 32)
+    throw new Error('illegal architecture specified (32 or 64).')
```

Comment:
I would still probably recommend using 'x64' and 'x86' and then convert them to just 64 and 86 when passing them to reg. It's nice to keep platform and architecture identifiers consistent for node libraries.